File: PyProject/leetcode/tree/binary-tree-inorder-traversal.py
from typing import List


# Definition for a binary tree node.
class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None


def helper(node: TreeNode, result: List[int]) -> List[int]:
    if not node:
        return result
    helper(node.left, result)
    result.append(node.val)
    helper(node.right, result)
    return result


class Solution:
    def inorderTraversal(self, root: TreeNode) -> List[int]:
        result = []
        return helper(root, result)

File: GoProject/leetcode/tree/binary-tree-inorder-traversal.go
package tree

/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

//func inorderTraversalOfHelper(root *TreeNode, res []int) []int {
//	if root == nil {
//		return res
//	}
//	res = inorderTraversalOfHelper(root.Left, res)
//	res = append(res, root.Val)
//	res = inorderTraversalOfHelper(root.Right, res)
//	return res
//}

func inorderTraversal(root *TreeNode) (res []int) {
	var inorder func(node *TreeNode)
	inorder = func(node *TreeNode) {
		if node == nil {
			return
		}
		inorder(node.Left)
		res = append(res, node.Val)
		inorder(node.Right)
	}
	inorder(root)
	return
}
